.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,3 +1,3 @@
 - [ ] Closes #xxxx
 - [ ] Tests added / passed
-- [ ] Passes `black distributed` / `flake8 distributed` / `isort distributed`
+- [ ] Passes `pre-commit run --all-files`
```

Comment:
cc @jakirkham @jacobtomlinson. This PR is proposing we switch to using `pre-commit` to run all code style checks instead of enumerating the individual checks that are made and documenting `pre-commit` as optional. I'm okay with this, but wanted to flag this for your thoughts as well